packages/ts-codegen/src/__tests__/index.test.ts

```diff
@@ -179,10 +179,14 @@ describe(generateTS, () => {
                 readonly userUpdated?: userUpdated<any>;
             }
             export type userUpdated<SubscribeResult = never> = {
+                /*We added this to fix the typescript narrowing problem.*/
+                kind?: "Subscribe";
                 subscribe: (model: unknown, args: {}, context: unknown, info: ResolveInfo) => PromiseOrValue<AsyncIterator<{
```

Comment:
These comments seem unnecessary, just have the commit/PR have a good description of the problem.